### ISSUE TITLE
Update IMA dependency to v3.8.2

### ISSUE THREAD

**Issue by victorraft** [google/ExoPlayer#3911]
Hi, I noticed that in the latest Exoplayer release (2.7.0) it still uses version 3.7.4 of the IMA SDK.
I tried to force an upgrade by defining a dependency on IMA 3.8.2 but I ran into trouble compiling. I was able to fix the compilation issues but then it just crashes when running.
It would be really helpful if you could update the IMA dependency to its latest version available. The company I work for really needs it.
Thank you!

**Comment by andrewlewis** [comment]
@victorraft Was the compilation issue an error like "More than one file was found with OS independent path [...]", or something else?

**Comment by victorraft** [comment]
@andrewlewis yes, exactly this kind of error. 
The errors were gone after I added a few `exclude` lines in my build.gradle, like this:
```
exclude 'META-INF/android.arch.lifecycle_runtime.version'
exclude 'META-INF/com.android.support_support-v4.version'
exclude 'META-INF/com.android.support_support-fragment.version'
exclude 'META-INF/com.android.support_support-media-compat.version'
exclude 'META-INF/com.android.support_support-core-utils.version'
exclude 'META-INF/com.android.support_support-compat.version'
exclude 'META-INF/com.android.support_support-core-ui.version'
```

Then, as I said before, it ran and crashed producing these logs:

`java.lang.IncompatibleClassChangeError: Found interface com.google.ads.interactivemedia.v3.api.ImaSdkSettings, but class was expected (declaration of 'com.google.ads.interactivemedia.v3.api.ImaSdkSettings' appears in /data/app/MY_APP_PACKAGE-zHX23MnIRu30bP9B3UQTaw==/base.apk!classes17.dex).`

From what I understand ImaSdkSettings was a class in IMA 3.7.4 and now is an interface in 3.8.2.

**Comment by andrewlewis** [comment]
Thanks for the information. I've followed up regarding the problematic version files [internal reference: b/74099519].

Having excluded them in the demo app build.gradle file I found that the extension worked as expected. Is it possible that you need to do a full rebuild of your project after changing the dependency? It may also be necessary to set `playServicesLibraryVersion = '11.8.0'` in constants.gradle.

**Comment by giladna** [comment]
@andrewlewis 
What is the reason this build error happening?
in case app uses 3rd parry libs that holds different playServicesLibraryVersion inside 
just upgrading my app to 11.8.0 does not help and seems like need to add the exclude lines.

any other option to fix it?
10x
Gilad.

**Comment by andrewlewis** [comment]
The reason for the error seems to be that `interactivemedia-3.8.2.jar` includes files that shouldn't be there. It doesn't actually expose the APIs of the libraries referred to in `META-INF`, but the files clash with those in a support library. Gradle gives up when it finds these duplicate files, as it doesn't know which one to choose.

The only way I know of to address this at the moment is to exclude the files in the APK packager as above, I'm afraid.

**Comment by giladna** [comment]
@andrewlewis 
Thanks!

**Comment by deeps2HOG** [comment]
All of a sudden, I am also getting this type of error. We don't use ExoPlayer but we use JWPlayer (which is built on top of ExoPlayer) to stream video. Although there is IMA dependency inside my build.gradle because of IMA ADs. 

I posted question on StackOverflow: 
https://stackoverflow.com/questions/49423572/android-unable-to-build-apk-more-than-one-file-was-found-with-os-independent-p/49424933#49424933
and after doing more googling, I found this GitHub issue.
I followed @victorraft solution of excluding some META-INFs and now I am able to build. 

**UPDATE**: Although Build was successful, but most of the media URLs are now throwing error: `Invalid HTTP response code: 404: Not Found.` As I am using JWPlayer so I have asked regarding this on their  repo: https://github.com/jwplayer/jwplayer-sdk-android-demo/issues/53
